src/Main.purs

```diff
@@ -1,15 +1,21 @@
 module Main where
 
 import Prelude
+
 import Component.Board as Board
 import Data.Location (initialLocations)
 import Effect (Effect)
+import Halogen as H
 import Halogen.Aff as HA
 import Halogen.VDom.Driver (runUI)
+import AppM (runAppM)
 
 main :: Effect Unit
 main = do
-  locations <- initialLocations
   HA.runHalogenAff do
+    locations <- runAppM initialLocations
     body <- HA.awaitBody
-    runUI Board.component locations body
+
+    let app = H.hoist runAppM Board.component
+    root <- runUI app locations body
```

Comment:
It's not currently used, I was looking into doing some stuff with coroutines and forgot to change it back before pushing.

It can just be `runUI app locations body` like you mentioned.